tests/unit/test_workflow_files.py

```diff
@@ -1553,44 +1571,6 @@ def test_remote_clean_cmd(
     assert constructed_cmd == ['clean', '--local-only', reg, *expected_args]
 
 
-@pytest.mark.parametrize(
-    'run_dir, srv_dir',
-    [
-        ('a', 'a/R/.service'),
-        ('d/a', 'd/a/a/R/.service'),
-        ('z/d/a/a', 'z/d/a/a/R/.service')
-    ]
-)
-def test_symlinkrundir_children_that_contain_workflows_raise_error(
-        run_dir, srv_dir, monkeypatch):
-    """Test that a workflow cannot be contained in a subdir of another
-    workflow."""
-    monkeypatch.setattr(
-        'cylc.flow.workflow_files.os.path.isdir',
-        lambda x: not (
-            x.find('.service') > 0 and x != srv_dir
-        )
-    )
-    monkeypatch.setattr(
-        'cylc.flow.workflow_files.get_cylc_run_abs_path',
-        lambda x: x)
-    monkeypatch.setattr(
-        'cylc.flow.workflow_files.os.scandir',
-        lambda x: [
-            mock.Mock(
-                path=srv_dir[0:len(x) + 2],
-                is_symlink=lambda: True
-            )
-        ]
-    )
-
-    try:
-        check_nested_run_dirs(run_dir, 'placeholder_flow')
-    except ServiceFileError:
-        pytest.fail(
-            "Unexpected ServiceFileError, Check symlink logic.")
```

Comment:
@datamel I have removed this test and replaced it with the one above. This test seemed to do the opposite of what it said it should? Surely it should have been using `with pytest.raises()`?